### PR TITLE
[Feature Store] Fix spark implementation of `DateExtractor`

### PR DESCRIPTION
**Pull request by gtopper**
[ML-3414](https://jira.iguazeng.com/browse/ML-3414)

### CODE REVIEW

File: mlrun/feature_store/steps.py [hunk 1]
```diff
@@ -505,16 +493,19 @@ def _do_pandas(self, event):
...
-        from pyspark.sql.functions import date_format
+        import pyspark.sql.functions
 
         for part in self.parts:
-            if part in self.fstore_date_format_to_spark_date_format:
+            func = part
+            if func == "day_of_year":
+                func = "dayofyear"
+            if func == "day_of_month":
+                func = "dayofmonth"
```
Comment (Tankilevitch) [inline mlrun/feature_store/steps.py:503]:
```suggestion
            if func in ("day_of_year", "day_of_month"):
                func = func.replace("_", "")
```

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: mlrun/feature_store/steps.py (hunk 1 above)
Comment (Tankilevitch) [reply mlrun/feature_store/steps.py:503]: or even take it to a helper function which will hold all those possible transformations and names

Comment (gtopper) [reply mlrun/feature_store/steps.py:503]: I was kind of on the fence about this because who knows what the user might put in there... But sure, why not.

Comment (gtopper) [reply mlrun/feature_store/steps.py]: Sure, why not.

File: mlrun/feature_store/steps.py [hunk 2]
```diff
@@ -505,16 +493,17 @@ def _do_pandas(self, event):
...
 
     def _do_spark(self, event):
-        from pyspark.sql.functions import date_format
+        import pyspark.sql.functions
 
         for part in self.parts:
-            if part in self.fstore_date_format_to_spark_date_format:
+            func = part
+            if func in ("day_of_year", "day_of_month"):
+                func = func.replace("_", "")
```
Comment (Tankilevitch) [inline mlrun/feature_store/steps.py:501]: why we do this trimming anyway? worth adding a comment IMO

**Review by Tankilevitch — APPROVED** [review]
Looks good, added a suggestion for better readability LMK what you think

File: tests/system/feature_store/test_spark_engine.py [hunk 3]
```diff
@@ -48,26 +48,29 @@
...
-def read_and_assert(csv_path_spark, csv_path_storey):
+def read_and_assert(out_path_spark, out_path_storey):
     read_back_df_spark = None
     file_system = fsspec.filesystem("v3io")
-    for file_entry in file_system.ls(csv_path_spark):
+    for file_entry in file_system.ls(out_path_spark):
         filepath = file_entry["name"]
         if not filepath.endswith("/_SUCCESS"):
-            read_back_df_spark = pd.read_csv(f"v3io://{filepath}")
+            read_back_df_spark = pd.read_parquet(f"v3io://{filepath}")
```
Comment (davesh0812) [inline tests/system/feature_store/test_spark_engine.py:57]: All the steps tests such as `test_ingest_with_steps_drop_features` using this function and pass to it csv path, I think you can change all the target to be parquet or to check here witch file path you get and use the right function : `read_csv` or `read_parquet`.

And also why did you change in the `DateExtractor` test the target to be parquet?

File: tests/system/feature_store/test_spark_engine.py [hunk 4]
```diff
@@ -48,26 +48,29 @@
...
             break
     assert read_back_df_spark is not None
 
     read_back_df_storey = None
-    for file_entry in file_system.ls(csv_path_storey):
+    for file_entry in file_system.ls(out_path_storey):
         filepath = file_entry["name"]
-        read_back_df_storey = pd.read_csv(f"v3io://{filepath}")
+        read_back_df_storey = pd.read_parquet(f"v3io://{filepath}")
```
Comment (davesh0812) [inline tests/system/feature_store/test_spark_engine.py:64]: same here

**Review by davesh0812 — COMMENTED** [review]
Just a small comment on the tests. :)

File: tests/system/feature_store/test_spark_engine.py (hunk 3 above)
Comment (gtopper) [reply tests/system/feature_store/test_spark_engine.py:57]: The question is why do they use CSV? The reason the test does not expose [ML-3414](https://jira.iguazeng.com/browse/ML-3414) is because CSV loses the type information.

I'll bring back the CSV implementation for exixting tests, but they should be migrated off of it if there is no good reason for them to be using CSV (such as if the purpose of the test is to test support for CSV).

Comment (davesh0812) [reply tests/system/feature_store/test_spark_engine.py]: Thank, good to know!

About the Implementation I think it is better to change all to parquet or at least not to write 2 equally functions... but it can be change in another pr I guess.

**Review by davesh0812 — APPROVED** [review]
